sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1003 @@
+        # super(LargeMarginNearestNeighbor, self).set_params(
+        # n_neighbors=self.n_neighbors_)
+
+        return X, y_inverse
+
+    def _init_transformer(self, X):
+        """Initialize the linear transformation by setting to user specified
+        parameter, loading from a file, applying PCA or setting to identity.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features_in)
+            Data samples.
+
+        Returns
+        -------
+        L : array, shape (n_features_out, n_features_in)
+            The initial linear transformation.
+        """
+
+        if self.L is not None:
+            L = np.asarray(self.L)
+        elif self.warm_start and hasattr(self, 'L_'):
+            L = self.L_
+        elif self.use_pca and X.shape[1] > 1:
+            pca = PCA(random_state=self.random_state_)
+            if self.verbose:
+                print('Finding principal components...', end='')
+                t = time.time()
+
+            pca.fit(X)
+
+            if self.verbose:
+                print('done in {:5.2f}s'.format(time.time() - t))
+
+            L = pca.components_
+        else:
+            L = np.eye(X.shape[1])
+
+        if self.n_features_out is None:
+            n_features_out = L.shape[0]
+        else:
+            n_features_out = self.n_features_out
+            if L.shape[0] > n_features_out:
+                warn('Decreasing the initial linear transformation output '
+                     'dimensionality ({}) to the preferred output '
+                     'dimensionality ({}).'.format(L.shape[0], n_features_out),
+                     DataDimensionalityWarning)
+                L = L[:n_features_out]
+
+        return L, n_features_out
+
+    def _select_target_neighbors(self, X, y):
+        """Find the target neighbors of each sample, that stay fixed during
+        training.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features_in)
+            The training samples.
+
+        y : array, shape (n_samples,)
+            The corresponding training labels indices.
+
+        Returns
+        -------
+        target_neighbors: array, shape (n_samples, n_neighbors)
+            An array of neighbors indices for each sample.
+        """
+
+        if self.verbose:
+            print('Finding the target neighbors...', end='')
+            t = time.time()
+
+        target_neighbors = np.empty((X.shape[0], self.n_neighbors_), dtype=int)
+
+        nn = NearestNeighbors(n_neighbors=self.n_neighbors_,
+                              algorithm=self.algorithm, n_jobs=self.n_jobs)
+
+        for class_id in self.classes_non_singleton_:
+            ind_class, = np.where(np.equal(y, class_id))
+            nn.fit(X[ind_class])
+            neigh_ind = nn.kneighbors(return_distance=False)
+            target_neighbors[ind_class] = ind_class[neigh_ind]
+
+        if self.verbose:
+            print('done in {:5.2f}s'.format(time.time()-t))
```

Comment:
Done. Weird that I didn't see that. I thought the tests also fail for PEP8 violations?